### ISSUE TITLE
`ocloc disasm` doesn't accept `-device` argument

### ISSUE THREAD

**Issue by mattst88** [intel/compute-runtime#582]
````sh
$ LD_LIBRARY_PATH=/usr/lib/llvm/14/lib64 ocloc compile -file icd-neo.txt-split/0x28308b77-1.cl -device dg2-g12 -output 0x28308b77-1.cl 
Build succeeded.
$ ocloc disasm -file 0x28308b77-1.cl_XE_HPG_COREdg2.bin -device dg2-g12 -dump dump               
Warning : missing or invalid -device parameter - results may be inacurate
Path to patch list not provided - using defaults, skipping patchokens as undefined.
Trying to disassemble main_function.krn
````

Why does `ocloc` print `Warning : missing or invalid -device parameter - results may be inacurate` despite the fact that I passed `-device dg2-g12` (which `ocloc compile` accepted!)?

**Comment by mattst88** [comment]
Maybe it just doesn't accept `dg2-g12`? `tgllp` seems to work fine:

````sh
$ ocloc compile -file icd-neo.txt-split/0x28308b77-1.cl -device tgllp -output 0x28308b77-1.cl 
Build succeeded.
$ ocloc disasm -file 0x28308b77-1.cl_Gen12LPlp.bin -device tgllp -dump dump                  
Path to patch list not provided - using defaults, skipping patchokens as undefined.
Trying to disassemble main_function.krn
````

**Comment by mattst88** [comment]
Any traction on this?

**Comment by daariahinz** [comment]
Yes, fix has been added in https://github.com/intel/compute-runtime/commit/31deb4fd63c101080ce11dd0f58274156b45c969

**Comment by JablonskiMateusz** [comment]
@mattst88 did it resolve your issue?

**Comment by mattst88** [comment]
Sorry, I  haven't been able to test it yet. The patch doesn't apply cleanly to the latest tag.

**Comment by mattst88** [comment]
I confirm this is fixed (tested 22.53.25242.13). Thanks!
